packages/omg-js-childchain/src/index.js

```diff
@@ -111,29 +74,86 @@ class ChildChain {
 
     return Array.from(balanceMap).map(elem => ({ currency: elem[0], amount: elem[1] }))
   }
-}
 
-function validateInputs (arg) {
-  // TODO
-  const valid = true
-  if (!valid) {
-    throw new InvalidArgumentError()
+  /**
+   * Create an unsigned transaction
+   *
+   * @method createTransaction
+   * @param {object} transactionBody
+   * @return {object}
+   */
+  createTransaction (transactionBody) {
+    transaction.validate(transactionBody)
+    const txArray = transaction.toArray(transactionBody)
+    return rlp.encode(txArray).toString('hex')
   }
-}
 
-function validateOutputs (arg) {
-  // TODO
-  const valid = true
-  if (!valid) {
-    throw new InvalidArgumentError()
+  /**
+   * Sign a transaction
+   *
+   * @method signTransaction
+   * @param {string} unsignedTx
+   * @param {array} privateKeys
+   * @return {array} signatures
+   */
+  signTransaction (unsignedTx, privateKeys) {
+    privateKeys.forEach(key => validatePrivateKey)
+    return sign(Buffer.from(unsignedTx, 'hex'), privateKeys)
   }
-}
 
-function validateCurrency (arg) {
-  // TODO
-  const valid = true
-  if (!valid) {
-    throw new InvalidArgumentError()
+  /**
+   * Build a signed transaction into the format expected by submitTransaction
+   *
+   * @method buildSignedTransaction
+   * @param {string} unsignedTx
+   * @param {array} signatures
+   * @return {object}
+   */
+  buildSignedTransaction (unsignedTx, signatures) {
+    // rlp-decode the tx bytes
+    const decodedTx = rlp.decode(Buffer.from(unsignedTx, 'hex'))
+    // Append the signatures
+    const signedTx = [...decodedTx, ...signatures]
+    // rlp-encode the transaction + signatures
+    return rlp.encode(signedTx).toString('hex')
+  }
+
+  /**
+   * Submit a signed transaction to the childchain
+   *
+   * @method submitTransaction
+   * @param {string} transaction
+   * @return {object}
+   */
+  async submitTransaction (transaction) {
+    // validateTxBody(transactionBody)
+    return submitTx(transaction, this.childChainUrl)
+  }
+
+  /**
+   * create, sign, build and submit transaction to the childchain using raw privatekey and single utxo
+   * 
+   * @method sendTransaction
+   * @param {object} utxo - (only supports single utxo)
```

Comment:
Why does this function only support one utxo?
`createTransaction()` supports multiple utxos and that's what we're calling anyway

Reply:
I was having some thoughts about this. I will make it allow for multiple utxos, I assume we can support 2 right now ?